metadata/src/main/java/org/apache/kafka/controller/metrics/QuorumControllerMetrics.java

```diff
@@ -157,6 +165,16 @@ public Long value() {
                 return newActiveControllers();
             }
         }));
+        registry.ifPresent(r -> r.newGauge(AVERAGE_IDLE_RATIO, new Gauge<Double>() {
+            @Override
+            public Double value() {
+                return avgIdleTimeRatio.measure(METRIC_CONFIG, time.milliseconds());
+            }
+        }));
+    }
+
+    public void updateIdleTime(long idleDurationMs) {
+        avgIdleTimeRatio.record(METRIC_CONFIG, (double) idleDurationMs, time.milliseconds());
```

Comment:
Why create a METRIC_CONFIG to have it ignored by TimeRatio? The issue is that TimeRatio implements an API specific for Kafka Metrics. One solution is to add a `measure` and `record` method to TimeRatio that doesn't require this metrics configuration.